accepted/2024/supplied-by-platform.md

```diff
@@ -0,0 +1,228 @@
+This avoids false positive by features such as NuGetAudit and other scanners that may be using the dependency graph.
+
+## Motivation
+
+In the early versions of .NET (Core), individual assemblies from the runtime were shipped as packages.
+Starting with .NET (Core) 3.0, targeting packs were used for delivering references assemblies. The reference assemblies are no longer represented in graphs the way Microsoft.NETCore.App, Microsoft.AspNetCore.App, and NETStandard.Library were.
+When a .NET (Core) 3.0 or later project depends on these older platform packages, we want them to be ignored in the graph, as those APIs are supplied by the platform via other mechanisms, such as targeting packs at build time and runtime packs for self-contained deployments.
+
+Even .NET 9, there are certain assemblies that ship as both packages, but are also part of certain shared frameworks such as ASP.NET. See details <https://github.com/dotnet/aspnetcore/issues/3609>. An example of such assembly/package is `Microsoft.Extensions.Logging`, which is [published on nuget.org](https://www.nuget.org/packages/Microsoft.Extensions.Logging/), but also part of the Microsoft.AspNetCore.App shared framework. Currently there's conflict resolution in the .NET SDK to ensure that the latest version is chosen.
+
+There are a few benefits:
+
+The fewer packages need to be downloaded, the better the performance of the restore algorithm will be.
+Beyond that, the extra packages within the graph, do make the resolution step more challenging. Some of the targeting packs used to bring in such a large package graph, that affected the resolution performance significantly. See  and <https://github.com/NuGet/Home/issues/11993> for more details. Furthermore, certain versions of popular packages have taken dependencies on these targeting packs for historical reasons. Such an example is <https://www.nuget.org/packages/log4net/2.0.10> which performs significantly better than <https://www.nuget.org/packages/log4net/2.0.9> when installed in a .NET (Core) projects, see <https://github.com/NuGet/Home/issues/10030>.
+
+Certain versions of these packages are no longer  part of the project graph, thus reducing the chances of false positives during scanning.
+This is especially important in the context of assemblies that are part of shared frameworks. The build time conflict resolution ensures that the most up to date version is used, despite the fact that the user referenced package may habe vulnerabilities.
+Examples: <https://github.com/dotnet/sdk/issues/30659#issuecomment-2072567192>.
+
+This changes significantly helps align what gets restored and what gets published. By avoiding packages that are simply not going to be used at runtime, what gets used at runtime becomes more transparent.
+
+## Explanation
+
+### Functional explanation
+
+A list of package id/versions to be pruned will be provided by the .NET SDK. The version will signify the highest version to be pruned.
```

Comment:
Today this information already ships in the individual targeting packs.  @dsplaisted doesn't think that NuGet nor the SDK can directly consume it from here since they will not be available prior to restore, so @dsplaisted was thinking that the SDK build would need to capture them and store locally - similar to what it does with "KnownFramework" information.